### PR TITLE
GH-78: keyboard guide

### LINKED ISSUE #78: add Guide functionality to the web app
create UI for keyboard guide following design in figma (https://www.figma.com/file/KT1vLyYfL28NZDdNeopdNj/re-design-hitt?node-id=737%3A3706&t=PFbmT1Bze64UWvXj-0)

### PR DESCRIPTION
**Pull request by alex-waverley**
added keyboard guide without animation
Screenshot 2023-03-23 at 10 59 02

Closes #78

### CODE REVIEW

File: frontend/src/components/guideCarousel/activeButtons/bodyElevetion.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { useTheme } from '@mui/material/styles'
+
+const BodyElevetion = () => {
```
Comment (sskorol) [inline frontend/src/components/guideCarousel/activeButtons/bodyElevetion.js:3]: Typo: Elevetion -> Elevation

File: frontend/src/components/guideCarousel/index.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+}))
+
+const guideItems = [
+  {
+    leggedRobotImage: robot,
+    title: 'Robot movement:',
+    activeButtonsDescription: RobotMovement,
+    keyboardSettings: {
+      template: 'normal',
+      actions: ['forward', 'backward', 'leftward-rotation', 'rightward-rotation'],
```
Comment (sskorol) [inline frontend/src/components/guideCarousel/index.js:66]: It probably makes sense to move actions into enum. I see dups and it's easy to make a mistake or forget to update something.

File: frontend/src/components/guideCarousel/index.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+  }
+}))
+
+const guideItems = [
+  {
+    leggedRobotImage: robot,
+    title: 'Robot movement:',
+    activeButtonsDescription: RobotMovement,
+    keyboardSettings: {
+      template: 'normal',
```
Comment (sskorol) [inline frontend/src/components/guideCarousel/index.js:65]: Seems like it's used in several places as well, so it makes sense to make it an enum.

File: frontend/src/components/guideCarousel/index.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+          {next && <>Next <ChevronRightIcon/></>}
+          {prev && <><ChevronLeftIcon/> Previous</>}
+      </Button>
+  )
+}
+
+const GuideCarousel = () => {
+  const onChange = (now, prev) => {
+    if(now !== prev) {
+      console.log('start animatino here')
```
Comment (sskorol) [inline frontend/src/components/guideCarousel/index.js:193]: Do you plan to add something here? + you have a typo in animation

File: frontend/src/components/guideCarousel/keyboardTemplate/index.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  const keyboardRef = useRef(null)
+  const Component = template === 'normal' ? NormalKeyboard : ShiftKeyboard
+  const { classes } = useStyles()
+
+  useEffect(() => {
+    actions
+      .filter(action => action !== current)
+      .forEach(action => keyboardRef.current.querySelector(`[data-action="${action}"]`)?.classList.add('active'))
+    
+    keyboardRef.current.querySelector(`[data-action="${current}"]`)?.classList.add('current')
```
Comment (sskorol) [inline frontend/src/components/guideCarousel/keyboardTemplate/index.js:41]: This line duplicates the one in forEach, the diff is only in 2 params.

File: frontend/src/components/guideModal/index.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        backgroundColor: theme.palette.background.guideModal,
+        padding: '34px',
+      }}>
+        <CloseButton title='Close Guide' onClick={onClose}/>
+        <InputTabs
+          tabStyles={{position: 'absolute', left: '48px', top: '30px'}}
+          renderKeyboardContent={() => (
+            <>
+              <TabPanelTitle>Keyboard Guide</TabPanelTitle>
+              <GuideCarousel />
```
Comment (sskorol) [inline frontend/src/components/guideModal/index.js:36]: I guess the component will be the same for joystick? The only diff will be in a content, right? Do you want to design this component with that knowledge in mind?

File: frontend/src/components/header/index.js [hunk 7]
```diff
@@ -83,15 +84,21 @@ const Header = observer(() => {
...
+  const [guideOpen, setGuideOpen] = useState(false);
 
   const handleChange = (_event, newValue) => {
     setActiveTab(newValue);
   };
 
-  const handleGuideClick = () => {
+  const handleGuideOpen = () => {
+    setGuideOpen(true)
     console.log('Guide button click')
```
Comment (sskorol) [inline frontend/src/components/header/index.js:95]: Do you still need this log here?

File: frontend/src/components/inputTabs/index.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    </Typography>
+  )
+}
+
+const InputTabs = observer(({ tabStyles = {position: 'absolute', right: '12px', top: '12px'}, renderKeyboardContent, renderJoystickContent }) => {
+  const [activeTab, setActiveTab] = useState(0)
+  const handleChange = (_event, newValue) => {
+    setActiveTab(newValue);
+  }
+  const keyboardContent = renderKeyboardContent()
```
Comment (sskorol) [inline frontend/src/components/inputTabs/index.js:54]: Just curious why do we need to put the result into variable. Wouldn't it work the same way while calling directly within tab panel block?

**Review by sskorol — CHANGES_REQUESTED** [review]

File: frontend/src/components/guideCarousel/activeButtons/bodyElevetion.js (hunk 1 above)
Comment (alex-waverley) [reply frontend/src/components/guideCarousel/activeButtons/bodyElevetion.js]: done

File: frontend/src/components/guideCarousel/index.js (hunk 4 above)
Comment (alex-waverley) [reply frontend/src/components/guideCarousel/index.js:206]: type fixed
it's a good place to add logic for animation on slide change

File: frontend/src/components/guideCarousel/keyboardTemplate/index.js (hunk 5 above)
Comment (alex-waverley) [reply frontend/src/components/guideCarousel/keyboardTemplate/index.js:41]: this part will be rewritten when animation for guide is ready

File: frontend/src/components/inputTabs/index.js (hunk 8 above)
Comment (alex-waverley) [reply frontend/src/components/inputTabs/index.js:54]: done

File: frontend/src/components/guideModal/index.js (hunk 6 above)
Comment (alex-waverley) [reply frontend/src/components/guideModal/index.js:36]: for now (without having design for joystick) I assume that both guides will use similar components like carousel and it's already separated. Let's keep it as it is for now.

File: frontend/src/components/guideCarousel/keyboardTemplate/index.js (hunk 5 above)
Comment (alex-waverley) [reply frontend/src/components/guideCarousel/keyboardTemplate/index.js:41]: this code will be rewritten when animation is ready

File: frontend/src/components/guideCarousel/index.js (hunk 4 above)
Comment (alex-waverley) [reply frontend/src/components/guideCarousel/index.js:206]: yes I do, it's a good place to handle animation logic on slide change

File: frontend/src/components/guideCarousel/index.js (hunk 3 above)
Comment (alex-waverley) [reply frontend/src/components/guideCarousel/index.js]: done

**Review by sskorol — APPROVED** [review]

**Pull request merged**